pandas/io/formats/style.py

```diff
@@ -1389,26 +1457,29 @@ def _background_gradient(
         text_color_threshold: float = 0.408,
         vmin: Optional[float] = None,
         vmax: Optional[float] = None,
+        gmap: Optional[Sequence] = None,
     ):
         """
-        Color background in a range according to the data.
+        Color background in a range according to the data or a gradient map
         """
-        if (
-            not isinstance(text_color_threshold, (float, int))
-            or not 0 <= text_color_threshold <= 1
-        ):
-            msg = "`text_color_threshold` must be a value from 0 to 1."
-            raise ValueError(msg)
-
+        if gmap is None:
+            gmap = s.to_numpy(dtype=float)
+        else:
+            try:
```

Comment:
hmm, shouldn't this just be a DataFrame that is aligned? why go to all of this complication (and its fragile)?

Reply:
In all cases the `gmap` has to match the data shape `s` if it doesn't it would just raise downstream error. This way, imo the code is simple and concise and provides both flexibility of input and helpful user error feedback.

in the case `axis` in [0,1] then `s`, the data, is guaranteed a 1d-series, and you can supply, as `gmap` either 1d or 2d inputs provided the second dim has length 1 so it can be reshaped:
 ```
df = DataFrame([[1,2],[3,4]])
...background_gradient(gmap=[2,4], axis=0)  # 1d list
...background_gradient(gmap=df[1], axis=0)  # series
...background_gradient(gmap=df[[1]], axis=0)  # dataframe
...background_gradient(gmap=[[2],[4]], axis=0)  # 2d list
...background_gradient(gmap=np.array([[2],[4]]), axis=0) # 2d numpy
...background_gradient(gmap=np.array([2,4]), axis=0)  # 1d numpy             ALL WORK
```

in the case `axis` is None and `s` is a DataFrame with 2 cols or more then you need a 2d input (or in edge-case a ravelled 1d input):
```
....background_gradient(gmap=[[2,2],[4,4]], axis=None)  # 2d list
...background_gradient(gmap=np.array([[2,2],[4,4]]), axis=None)  # 2d numpy
...background_gradient(gmap=df, axis=None)  # dataframe
...background_gradient(gmap=[2,2,4,4], axis=None)  # 1d edge case.
```

In the case `axis` is None and the the data is a 1-col dataframe any of the 1d cases will again work, e.g
```
.background_gradient(gmap=[2,4], subset=[0], axis=None)
```

Reply:
you need to for sure align if a dataframe is passed.
ok on 2D. why is accepting 1D ok ? this is just too flexible. I dont' see any advantage to this.

Reply:
Ok how's this: I've split the 1d and 2d cases still compare shape at end with error if they don't match.